<launch>
  <arg name="camera_frame" default="/kinect2_rgb_optical_frame" />
  <arg name="pddl_solver_topic" default="/pddl_planner" />

  <arg name="kinect2" default="false"/>
  <arg if="$(arg kinect2)" name="point_cloud_topic" default="/kinect2/sd/points" />
  <arg if="$(arg kinect2)" name="depth_topic" default="/kinect2/sd/image_depth" />
  <arg if="$(arg kinect2)" name="color_topic" default="/kinect2/sd/image_depth_rect" />

  <arg name="kinect" default="false"/>
  <arg if="$(arg kinect)" name="point_cloud_topic" default="/camera/depth_registered/points" />
  <arg if="$(arg kinect)" name="depth_topic" default="/camera/depth_registered/image_raw" />
  <arg if="$(arg kinect)" name="color_topic" default="/camera/rgb/image_rect_color" />

  <arg name="sim" default="false" />
  <arg if="$(arg sim)" name="point_cloud_topic" default="/mock_point_cloud" />
  <arg if="$(arg sim)" name="depth_topic" default="/camera/depth_registered/image_raw" />
  <arg if="$(arg sim)" name="color_topic" default="/camera/rgb/image_rect_color" />

  <!-- start camera -->
  <include file="$(find rapid_pbd)/launch/kinect2.launch" unless="$(arg sim)">
    <arg name="kinect" value="$(arg kinect)" />
    <arg name="kinect2" value="$(arg kinect2)" />
  </include>

  <group ns="rapid_pbd"> 
    <param name="robot" value="baxter" />
    <node pkg="rapid_pbd" type="baxter_action_node" name="baxter_actuator_server" />
    <node pkg="rapid_pbd" type="program_execution_node" name="program_executor">
      <remap from="l_arm_joint_action" to="/robot/limb/left/follow_joint_trajectory" />
      <remap from="r_arm_joint_action" to="/robot/limb/right/follow_joint_trajectory" />
      <remap from="move_group" to="/move_group" />
    </node>
    <rosparam command="load" file="$(find rapid_pbd)/config/surface_segmentation.yaml" />
    <node pkg="rapid_pbd" type="surface_segmentation_node" name="surface_segmentation_node" args="$(arg point_cloud_topic)" />
    <node pkg="rapid_pbd" type="pddl_solver_node" name="pddl_solver_node" args="$(arg pddl_solver_topic)"/>

    <rosparam command="load" file="$(find rapid_pbd)/config/planning_domain.yaml" />
  </group>

  <!-- Depth cloud -->
  <node pkg="nodelet" type="nodelet" name="rapid_pbd_convert_metric" args="standalone depth_image_proc/convert_metric" if="$(arg sim)">
    <remap from="image_raw" to="$(arg depth_topic)" />
    <remap from="image" to="$(arg depth_topic)_float" />
  </node>
  <node pkg="depthcloud_encoder" type="depthcloud_encoder_node" name="rapid_pbd_depthcloud_encoder">
    <param name="rgb" value="$(arg color_topic)" />
    <param if="$(arg sim)" name="depth" value="$(arg depth_topic)" />
    <param unless="$(arg sim)" name="depth" value="$(arg depth_topic)_float" />
    <param name="camera_frame_id" value="$(arg camera_frame)" />
  </node>
</launch>
